### DISCUSSION TITLE
Using Swap and a Dropdown at the same time

### DISCUSSION THREAD

**Discussion by ridgey28** [saadeghi/daisyui#2292] [General]
Hi. I'm just wondering if it is possible to use a swap which activates a dropdown and swaps the icon? 

Example; like in a hamburger menu which will switch to an X when open? I know I can do this with JS but is it possible with CSS too? I'm guessing the checkbox is an issue here with CSS and the click outside for the menu to disappear would also prevent this unless using JS.

I'm loving daisy UI so far. Many thanks

**Comment by saadeghi** [comment]
It is possible and it is something I'm going to add on the next update of `swap` style.  
Right now I'm really busy with fixing issues. I will get into it as soon as I get some free time.  

For now, if you don't want to use JS, you can do something like this:  
https://play.tailwindcss.com/brzQpy3SWo

**Reply by fnuttens** [reply]
Hey! Kudos for making daisyUI, it has saved me a huge amount of time and effort ❤️

Have you had time to look into this?

**Comment by egormkn** [comment]
I have made another variant of dropdown with swap button using only CSS. Maybe someone will find it useful:

```jsx
<div className="daisy-dropdown group">
  <label tabIndex={0} role="button" className="daisy-btn daisy-btn-circle daisy-swap daisy-swap-rotate group-focus-within:daisy-swap-active">
    <svg
      className="daisy-swap-off group-focus-within:-rotate-45 fill-current"
      xmlns="http://www.w3.org/2000/svg"
      width="32"
      height="32"
      viewBox="0 0 512 512">
      <path d="M64,384H448V341.33H64Zm0-106.67H448V234.67H64ZM64,128v42.67H448V128Z" />
    </svg>

    <svg
      className="daisy-swap-on group-focus-within:rotate-none fill-current"
      xmlns="http://www.w3.org/2000/svg"
      width="32"
      height="32"
      viewBox="0 0 512 512">
      <polygon
        points="400 145.49 366.51 112 256 222.51 145.49 112 112 145.49 222.51 256 112 366.51 145.49 400 256 289.49 366.51 400 400 366.51 289.49 256 400 145.49" />
    </svg>
  </label>
  <ul tabIndex={-1} className="daisy-dropdown-content daisy-menu bg-base-100 rounded-box z-1 w-52 p-2 shadow-sm">
    <li><a>Item 1</a></li>
    <li><a>Item 2</a></li>
  </ul>
</div>
```

**Reply by pdanpdan** [reply]
A play for above solution: https://play.tailwindcss.com/H3pM0ZQCER
